src/routes/pay/likecoin.ts

```diff
@@ -215,14 +215,25 @@ likecoinRouter.post('/', async (req, res, next) => {
       throw new Error('callback has no "metadata"')
     }
 
-    const trans = (
+    const txState = metadata.state || ''
+    logger.info(`callback tx:${txHash}, state: ${txState}`)
+
+    let trans = (
       await paymentService.findTransactions({
         providerTxId: tx.txHash,
       })
     )[0]
 
     if (!trans) {
-      throw new Error(`counld not find tx hash`)
+      trans = (
+        await paymentService.findTransactions({
+          providerTxId: txState,
+        })
+      )[0]
+
+      if (!trans) {
+        throw new Error(`counld not find tx hash`)
```

Comment:
may define a custom error type in `src/common/errors.ts` and use it here?   will be better to find it in Sentry for troubleshooting?

Reply:
@tx0c good suggestion. let me review the whole file and refactor the error handling accordingly after this fix.